docs/DeveloperGuide.adoc

```diff
@@ -302,27 +306,126 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+==== Design Considerations
+
+===== Aspect: How sell command is implemented
+
+* **Alternative 1 (current choice):** Decreases quantity in the Quantity Class.
+** Pros: Code is more cohesive.
+** Cons: Adds more code to Quantity Class
+* **Alternative 2:** Decreases quantity in the Book Class.
+** Pros: Does not need to edit Quantity Class.
+** Cons: Code becomes less cohesive.
+// end::sellcommand[]
+
+// tag::Request[]
+=== Request feature
+==== Current implementation
+The Request command utilises both the `RequestModel` and `Logic` component to fulfil its function.
+This is accessible for all users.
+
+==== Design Considerations
+===== Aspect: Request command is implemented in a different category
+As mentioned above, Request is called in RequestModel and RequestStorage.
+A request object consists of three objects, Isbn, Email, and Quantity.
+
+===== RequestModel component
+
+.Structure of the RequestModel Component
+image::RequestModelClassDiagram.jpg[width="800"]
+
+The implementation is similar to 2.4. Model component.
+Note that XmlAdaptedTag class is removed as request has no tags.
+
+===== RequestStorage component
+
+.Structure of the RequestStorage Component
+image::RequestStorageClassDiagram.jpg[width="800"]
+
+Likewise, this implementation is similar to 2.5. Storage component.
+
+===== Aspect: How Request command is implemented
+
+.Structure of Request (High Level Sequence Diagram)
+image::HighLevelRequestSequenceDiagram.png[width="800"]
+
+* **Similar to Add Command**
+** After request command is called by the user, it gets parsed through a RequestListParser.
+** The UI also prompts to EventCenter and logs out that the Request List has changed.
+** Request object is created and added to the model.
+** The storage then receives the request and returns message to model, which is then transferred to the UI.
+** This notifies the user that the request is successfully submitted.
+
+// end::Request[]
+
+
+// tag::stockcommand[]
+=== Stock feature
+==== Current implementation
+The sell command utilises both the `Model` and `Logic` component to fulfil its function.
+
+The stock operation is similar to the sell operation, refer to sell command for sequence diagram:
+
+==== Design Considerations
+
+===== Aspect: How stock command is implemented
+
+* **Alternative 1 (current choice):** Increases quantity in the Quantity Class.
+** Pros: Code is more cohesive.
+** Cons: Adds more code to Quantity Class.
+* **Alternative 2:** Replace quantity in the Book Class.
+** Pros: Does not need to edit Quantity Class.
+** Cons: Code becomes less cohesive.
+// end::stockcommand[]
+
 // tag::dataencryption[]
-=== [Proposed] Data Encryption
+=== Password Protection
 
-_{Explain here how the data encryption feature will be implemented}_
+==== Current Implementation
```

Comment:
Maybe you can write a bit more about some ways you have kept the password secure. Some diagrams to show interaction between classes when using passwords would help :)